sktime/transformations/series/dropna.py

```diff
@@ -0,0 +1,239 @@
+        else:
+            self._remember = self.remember
+
+    def _check_axis(self, axis):
+        """Check axis parameter, should be a valid string as per docstring."""
+        if axis not in self.VALID_AXIS_VALUES:
+            raise ValueError(
+                f'invalid axis parameter value encountered: "{axis}", '
+                f"axis must be one of: {self.VALID_AXIS_VALUES}"
+            )
+
+        return axis
+
+    def _check_how(self, how):
+        """Check how parameter, should be a valid string as per docstring."""
+        if how not in self.VALID_HOW_VALUES:
+            raise ValueError(
+                f'invalid how parameter value encountered: "{how}", '
+                f"how must be one of: {self.VALID_HOW_VALUES}"
+            )
+
+        return how
+
+    def _check_thresh(self, thresh, how):
+        """Check thresh parameter, should be a valid value as per docstring."""
+        if not isinstance(thresh, self.VALID_THRESH_TYPES):
+            raise ValueError(
+                f'invalid thresh parameter value encountered: "{thresh}", '
+                f"thresh must be of type: {self.VALID_HOW_VALUES}"
+            )
+        if isinstance(thresh, float) and not (0 < thresh < 1):
+            raise ValueError("thresh must be a fraction between zero and one")
+        if (how is not None) and (thresh is not None):
+            raise TypeError("thresh cannot be set together with how")
+
+        return thresh
+
+    def _check_remember(self, remember):
+        """Check remember parameter, should be a valid type as per docstring."""
+        if not isinstance(remember, self.VALID_REMEMBER_TYPES):
+            raise ValueError(
+                f'invalid remember parameter value encountered: "{remember}", '
+                f"remember must be of type: {self.VALID_REMEMBER_TYPES}"
+            )
+
+        return remember
+
+    def _fit(self, X, y=None):
+        """Fit transformer to X and y.
+
+        private _fit containing the core logic, called from fit
+
+        Parameters
+        ----------
+        X : pd.DataFrame
+            if self.get_tag("univariate-only")==True:
+                guaranteed to have a single column
+            if self.get_tag("univariate-only")==False: no restrictions apply
+        y : None, present only for interface compatibility
+
+        Returns
+        -------
+        self: reference to self
+        """
+        self.dropped_index_values_ = None
+        self._agg_axis = 1 - self._axis
+
+        mask = None
+        if self._how == "any":
+            mask = X.isna().any(axis=self._agg_axis)
+        elif self._how == "all":
+            mask = X.isna().all(axis=self._agg_axis)
+        elif isinstance(self._thresh, int):
+            mask = X.count(axis=self._agg_axis) < self._thresh
+        elif isinstance(self._thresh, float):
+            mask = (
+                X.count(axis=self._agg_axis).div(X.shape[self._agg_axis]) < self._thresh
+            )
```

Comment:
Why not pass this task to `pandas`?

```suggestion
        mask = X.dropna(axis=self._agg_axis, how=self._how, thresh=self._thresh)
```